webauthn-authenticator-rs/src/macos.rs

```diff
@@ -0,0 +1,57 @@
+use crate::{AuthenticatorBackend, Url, WebauthnCError};
+use webauthn_rs_proto::{
+    PublicKeyCredential, PublicKeyCredentialCreationOptions, PublicKeyCredentialRequestOptions,
+    RegisterPublicKeyCredential,
+};
+
+mod sys {
+    use swift_rs::{swift, SRString};
+
+    swift!(pub fn perform_register(options: SRString) -> SRString);
+    swift!(pub fn perform_auth(options: SRString) -> SRString);
+
+    #[derive(serde::Deserialize)]
+    pub enum Result<T> {
+        #[serde(rename = "data")]
+        Data(T),
+        #[serde(rename = "error")]
+        Error(String),
+    }
+}
+
+/// Authenticator backend for MacOS ASAuthorization API.
+#[derive(Default)]
+pub struct MacOS {}
+
+impl AuthenticatorBackend for MacOS {
+    /// Perform a registration action using the ASAuthorization API.
+    fn perform_register(
+        &mut self,
+        _origin: Url,
```

Comment:
Because macOS is using the RP ID for both the `origin` field of the Client Data JSON *and* `MakeCredentialRequest.(RelyingParty)rp.id`, and doesn't have a way to pass other `RelyingParty` fields, I feel there should be an explicit check here.

Windows passes them as two separate fields, and I think our native CTAP2 implementation doesn't actually check it.